### PR TITLE
chore: cleanup as unknown conversion

### PR DESCRIPTION
**Pull request by zhaoyongjie**
### SUMMARY

Clean up _as unknown_ after monorepo

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

N/A

### TESTING INSTRUCTIONS

Pass all CI

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/packages/superset-ui-core/test/query/buildQueryObject.test.ts [hunk 1]
```diff
@@ -285,7 +285,8 @@ describe('buildQueryObject', () => {
         datasource: '5__table',
         granularity_sqla: 'ds',
         viz_type: 'table',
-        url_params: null as unknown as undefined,
+        // @ts-ignore
```
Comment (zhaoyongjie) [inline superset-frontend/packages/superset-ui-core/test/query/buildQueryObject.test.ts:288]: ignore is used for the test case.

File: superset-frontend/src/SqlLab/components/QuerySearch/index.tsx [hunk 2]
```diff
@@ -226,7 +226,7 @@ function QuerySearch({ actions, displayLimit }: QuerySearchProps) {
               value: xt,
               label: xt,
             }))}
-            value={from as unknown as undefined}
+            value={{ value: from, label: from }}
```
Comment (zhaoyongjie) [inline superset-frontend/src/SqlLab/components/QuerySearch/index.tsx:229]: bycatch: use **object** fill in Select value

File: superset-frontend/src/SqlLab/components/QuerySearch/index.tsx [hunk 3]
```diff
@@ -235,7 +235,7 @@ function QuerySearch({ actions, displayLimit }: QuerySearchProps) {
             name="select-to"
             placeholder={t('[To]-')}
             options={TIME_OPTIONS.map(xt => ({ value: xt, label: xt }))}
-            value={to as unknown as undefined}
+            value={{ value: to, label: to }}
```
Comment (zhaoyongjie) [inline superset-frontend/src/SqlLab/components/QuerySearch/index.tsx:238]: same before

File: superset-frontend/src/views/CRUD/data/database/DatabaseModal/index.tsx [hunk 4]
```diff
@@ -371,11 +371,11 @@ function dbReducer(
...
       ) {
-        const engineParamsCatalog = Object.keys(
+        const engineParamsCatalog = Object.entries(
           extra_json?.engine_params?.catalog || {},
-        ).map(e => ({
-          name: e,
-          value: extra_json?.engine_params?.catalog[e],
+        ).map(([key, value]) => ({
+          name: key,
+          value,
```
Comment (zhaoyongjie) [inline superset-frontend/src/views/CRUD/data/database/DatabaseModal/index.tsx:378]: same logical substitution

File: superset-frontend/src/views/CRUD/data/database/types.ts [hunk 5]
```diff
@@ -79,7 +79,7 @@ export type DatabaseObject = {
   // Extra
   extra_json?: {
     engine_params?: {
-      catalog: Record<any, any> | string;
+      catalog?: Record<any, any> | string;
```
Comment (zhaoyongjie) [inline superset-frontend/src/views/CRUD/data/database/types.ts:82]: `catalog` is a optional field.

File: superset-frontend/packages/superset-ui-core/src/color/utils.ts [hunk 6]
```diff
@@ -75,7 +75,16 @@ export function getAnalogousColors(colors: string[], results: number) {
 }
 
 export function addAlpha(color: string, opacity: number): string {
-  // coerce values so ti is between 0 and 1.
```
Comment (zhaoyongjie) [inline superset-frontend/packages/superset-ui-core/src/color/utils.ts:78]: bycatch: fix codes coverage.

File: superset-frontend/packages/superset-ui-core/src/color/utils.ts [hunk 7]
```diff
@@ -75,7 +75,16 @@ export function getAnalogousColors(colors: string[], results: number) {
...
+    throw new Error(
+      `The alpha channel should between 0 and 1, but got: ${opacity}`,
+    );
+  }
+  // the alpha value is between 00 - FF
+  const alpha = `0${Math.round(opacity * 255)
+    .toString(16)
+    .toUpperCase()}`.slice(-2);
+
+  return `${color}${alpha}`;
```
Comment (zhaoyongjie) [inline superset-frontend/packages/superset-ui-core/src/color/utils.ts:87]: bycatch: fix codes coverage.

File: superset-frontend/packages/superset-ui-core/test/query/buildQueryObject.test.ts (hunk 1 above)
Comment (villebro) [reply superset-frontend/packages/superset-ui-core/test/query/buildQueryObject.test.ts:288]: I believe `// @ts-expect-error` is the correct syntax for testing with incorrect types

**Review by villebro — APPROVED** [review]
Thanks for making the code cleaner! One nit, other than that LGTM

**Pull request merged**